src/Compilers/Shared/BuildServerConnection.cs

```diff
@@ -72,7 +75,7 @@ internal static BuildRequest CreateBuildRequest(
                 arguments,
                 workingDirectory: workingDirectory,
                 tempDirectory: tempDirectory,
-                compilerHash: BuildProtocolConstants.GetCommitHash() ?? "",
+                compilerHash: compilerHash ?? BuildProtocolConstants.GetCommitHash() ?? "",
```

Comment:
How is it intended for the SDK to get the compiler hash?

Reply:
Similarly to how `BuildProtocolConstants.GetCommitHash` would do it (look for CommitHashAttribute on a compiler dll). `BuildProtocolConstants.GetCommitHash` itself didn't work because this is a source package so the assembly is not the C# assembly but now I realized I can probably make that utility work by changing the assembly it looks at and then remove this parameter.